### PR TITLE
[ERP-6532] Main Ledger changes for wallet

### PR DESCRIPTION
**Pull request by amantheshooter**

### CODE REVIEW

File: src/main/java/lib/dehaat/ledger/presentation/RevampLedgerViewModel.kt [hunk 1]
```diff
@@ -90,6 +94,21 @@ class RevampLedgerViewModel @Inject constructor(
...
+            is APIResultEntity.Success -> {
+                viewModelState.update {
+                    it.copy(walletBalance = response.data?.totalWalletBalance ?: 0.0)
+                }
+            }
+            is APIResultEntity.Failure -> {
+                sendFailureEvent(response.getApiFailureError())
+            }
+        }
+        callingAPI()
```
Comment (lovemhjn) [inline src/main/java/lib/dehaat/ledger/presentation/RevampLedgerViewModel.kt:109]: rename this

File: src/main/java/lib/dehaat/ledger/presentation/ledger/revamp/state/transactions/ui/TransactionsScreen.kt [hunk 2]
```diff
@@ -273,6 +294,52 @@ fun TransactionsScreen(
...
+					)
+				})
+			LaunchedEffect(Unit) {
+				ledgerViewModel.ledgerAnalytics.onHoldAmountWidgetViewed()
+			}
+		}
+		EntryPointButtonContent(modifier = Modifier
+			.weight(.5f)
+			.padding(start = 8.dp, end = 8.dp),
+			valueText = walletBalance.getAmountInRupees(),
```
Comment (lovemhjn) [inline src/main/java/lib/dehaat/ledger/presentation/ledger/revamp/state/transactions/ui/TransactionsScreen.kt:335]: move walletBalance.getAmountInRupees() to view mapper class

File: src/main/res/drawable/ic_lock_orange_bg.xml [hunk 3]
```diff
@@ -3,12 +3,10 @@
...
     android:viewportWidth="32"
     android:viewportHeight="32">
-  <path
-      android:pathData="M0,0h32v32h-32z"
-      android:fillColor="#ffffff"/>
+
   <path
       android:pathData="M16,0L16,0A16,16 0,0 1,32 16L32,16A16,16 0,0 1,16 32L16,32A16,16 0,0 1,0 16L0,16A16,16 0,0 1,16 0z"
-      android:fillColor="#FFEEE6"/>
+      android:fillColor="#FFFFFF"/>
```
Comment (lovemhjn) [inline src/main/res/drawable/ic_lock_orange_bg.xml:9]: Check if this doesn't override any existing usage

**Review by lovemhjn — CHANGES_REQUESTED** [review]

**Review by sachin-varshney — APPROVED** [review]

File: src/main/java/lib/dehaat/ledger/navigation/LedgerNavigation.kt [hunk 4]
```diff
@@ -94,6 +96,21 @@ fun LedgerNavigation(
...
+        navBaseComposable(
+            route = LedgerRoutes.WalletLedgerRoute.screen,
+            arguments = listOf(navArgument(LedgerConstants.KEY_PARTNER_ID) {
+                type = NavType.StringType
+                defaultValue = partnerId
+            }),
+            logScreenName = ledgerCallbacks.firebaseScreenLogger
+        ) {
+            WalletLedgerScreen(onClick = {
+	            openOrderDetailFragment(it, it)
```
Comment (shivam21) [inline src/main/java/lib/dehaat/ledger/navigation/LedgerNavigation.kt:108]: send only orderId LedgerExecutor already have partnerId

File: src/main/java/lib/dehaat/ledger/presentation/RevampLedgerViewModel.kt (hunk 1 above)
Comment (amantheshooter) [reply src/main/java/lib/dehaat/ledger/presentation/RevampLedgerViewModel.kt:152]: this is old function of ledger view model, I have just used it, Currently not doing it.

**Pull request merged**